thumbor/src/thumbor_/loaders/generate_object_url.py

```diff
@@ -1,20 +1,24 @@
-import loaders.bucket_details as bucket_details
-import boto3
-from botocore.client import Config
+import os
 import urllib.parse
 
-s3_buckets = boto3.resource(
-        service_name="s3",
-        endpoint_url=bucket_details.ep_url,
-        aws_access_key_id=bucket_details.key_id,
-        aws_secret_access_key=bucket_details.access_key,
-        config=Config(signature_version="s3v4"),
-        )
+import aioboto3
+from botocore.config import Config
 
-my_bucket = s3_buckets.Bucket(bucket_details.bucket_name)
-client = s3_buckets.meta.client
 
-def generate_encoded_url(input_key):
-    generated_url = client.generate_presigned_url("get_object", ExpiresIn=3600, Params={"Bucket":bucket_details.bucket_name, "Key":input_key})
-    encoded_url = urllib.parse.quote(generated_url)
-    return encoded_url
\ No newline at end of file
+async def generate_encoded_url(input_key):
+    session = aioboto3.Session()
+    async with session.client(
+        "s3",
+        endpoint_url=os.environ.get("EP_URL"),
+        aws_access_key_id=os.environ.get("KEY_ID"),
+        aws_secret_access_key=os.environ.get("ACCESS_KEY"),
```

Comment:
Could we check that there are set before trying to use them in the client? (and raise useful errors if they are not)

Reply:
Perhaps this can go to the backlog along with doing the check once then passing them on as context